mol-loader/Pubchem.py

```diff
@@ -0,0 +1,69 @@
+"""
+Encapsulate reading and formatting of PubChem
+"""
+
+from Sdf import Sdf
+import json
+import os, os.path
+import rdkit.Chem
+
+class Pubchem(Sdf):
+    def __init__(self, path):
+        Sdf.__init__(self, path)
+
+    SYNONYM_PROPS = [
+        'PUBCHEM_IUPAC_OPENEYE_NAME',
+        'PUBCHEM_IUPAC_CAS_NAME',
+        'PUBCHEM_IUPAC_NAME',
+        'PUBCHEM_IUPAC_SYSTEMATIC_NAME',
+        'PUBCHEM_IUPAC_TRADITIONAL_NAME'
+    ]
+    SMILES_PROPS = [
+        'PUBCHEM_OPENEYE_CAN_SMILES',
+        'PUBCHEM_OPENEYE_ISO_SMILES'
+    ]
+
+    def mol_to_dict(self, mol):
+        """
+        Capture all the information from a PubChem molecule into a dictionary
+
+        mol_to_dict( (Pubchem)self, (rdkit.Chem.Mol)mol) -> dict
+        """
+        props = set(mol.GetPropNames())
+        # All the properties defined in the molecule, as-is
+        d = {p.lower() : self.cast(mol.GetProp(p)) for p in props}
+        d['id'] = 'https://pubchem.ncbi.nlm.nih.gov/compound/' + mol.GetProp('PUBCHEM_COMPOUND_CID')
+
+        smiles = [mol.GetProp(p) for p in self.SMILES_PROPS if p in props]
+        smiles.append(rdkit.Chem.MolToSmiles(mol)) # Original SMILES
+        # Attempt normalization; this may fail if there are ... oddities ...
+        # in the molecule.
+        rdkit_mol = rdkit.Chem.MolFromSmiles(rdkit.Chem.MolToSmiles(mol)) # Canonicalizes
+        if rdkit_mol is not None:
+            d['rdkit_smiles'] = rdkit.Chem.MolToSmiles(rdkit_mol)
```

Comment:
`Chem.MolToSmiles()` has bad defaults, you want: `Chem.MolToSmlies(mol,True)`
so that the output SMILES includes information about stereochemistry. This should be done everywhere you construct SMILES.